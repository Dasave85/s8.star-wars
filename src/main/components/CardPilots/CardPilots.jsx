export const CardPilots = ({ pilot }) => {
  return (
    <div className="col-6">
      <div className="card m-5 border-2 " style={{ width: "25rem" }}>
        <div className="card-body">
          <h5 className="card-title my-3">Pilot: {pilot.name}</h5>
          <hr />
          <p className="card-text col-6">Height: {pilot.height}</p>
          <p className="card-text col-6">Birth Year: {pilot.birth_year}</p>
          <p className="card-text col-6">Mass: {pilot.mass}</p>
          <p className="card-text col-6">Hair Color: {pilot.hair_color}</p>
        </div>
      </div>
    </div>
  );
};
