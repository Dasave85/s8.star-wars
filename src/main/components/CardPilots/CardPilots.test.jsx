import { render, screen } from "@testing-library/react";
import { CardPilots } from "./CardPilots";

describe("Pruebas en el componente <CardPilot />", () => {
  const pilot = {
    name: "James",
    height: "1.73",
    birth_year: "25/06/1985",
    mass: "85",
    hair_color: "brown",
  };
  test("debe de hacer match con el snapshot", () => {
    const { container } = render(<CardPilots pilot={pilot} />);
    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar la informacion de un piloto", () => {
    render(<CardPilots pilot={pilot} />);

    expect(screen.getByText("Pilot: James")).toBeTruthy();
    expect(screen.getByText("Height: 1.73")).toBeTruthy();
    expect(screen.getByText("Birth Year: 25/06/1985")).toBeTruthy();
    expect(screen.getByText("Hair Color: brown")).toBeTruthy();
  });
});
