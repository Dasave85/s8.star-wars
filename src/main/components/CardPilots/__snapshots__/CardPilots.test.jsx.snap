// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <CardPilot /> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="col-6"
  >
    <div
      class="card m-5 border-2 "
      style="width: 25rem;"
    >
      <div
        class="card-body"
      >
        <h5
          class="card-title my-3"
        >
          Pilot: 
          James
        </h5>
        <hr />
        <p
          class="card-text col-6"
        >
          Height: 
          1.73
        </p>
        <p
          class="card-text col-6"
        >
          Birth Year: 
          25/06/1985
        </p>
        <p
          class="card-text col-6"
        >
          Mass: 
          85
        </p>
        <p
          class="card-text col-6"
        >
          Hair Color: 
          brown
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Pruebas en el componente <CardPilot /> debe de mostrar la informacion de un piloto 1`] = `
<div>
  <div
    class="col-6"
  >
    <div
      class="card m-5 border-2 "
      style="width: 25rem;"
    >
      <div
        class="card-body"
      >
        <h5
          class="card-title my-3"
        >
          Pilot: 
          James
        </h5>
        <hr />
        <p
          class="card-text col-6"
        >
          Height: 
          1.73
        </p>
        <p
          class="card-text col-6"
        >
          Birth Year: 
          25/06/1985
        </p>
        <p
          class="card-text col-6"
        >
          Mass: 
          85
        </p>
        <p
          class="card-text col-6"
        >
          Hair Color: 
          brown
        </p>
      </div>
    </div>
  </div>
</div>
`;
